string data
---
int64 magnet